app/controllers/index.js

```diff
@@ -54,11 +56,28 @@ export default class IndexController extends Controller {
     });
   }
 
+  async fetchAdditionalUserDetails(users) {
+    let userPromises = [];
+    users.forEach((user) => {
+      userPromises.push(fetch(user.url).then((response) => response.json()));
+    });
+
+    let userDataResponses = await all(userPromises);
+
+    return users.map((user, index) => ({
+      ...user,
+      name: userDataResponses[index].name || "",
+    }));
+  }
+
   updateContributorsList(users) {
     const contributorsList = users
       .map(user => {
-        const { handle, profileLink } = user;
-        return `<a href="${profileLink}" rel="noopener noreferrer" target="_blank">${handle}</a>`;
+        const { handle, profileLink, name } = user;
+
+        const constructedName = name ? `${name} ` : '';
+
+        return `<a href="${profileLink}" rel="noopener noreferrer" target="_blank">${constructedName}(${handle})</a>`;
```

Comment:
**suggestion (soft-blocking):**

When `name` doesn't exist, I think showing the handle without parentheses may look better.

Can we use an if statement to build display name?

```suggestion
        let displayName;

        if (name) {
          displayName = `${name} (${handle})`;
        } else {
          displayName = handle;
        }

        return `<a href="${profileLink}" rel="noopener noreferrer" target="_blank">${displayName}</a>`;
```

Reply:
Yeah, we can!